traits-stubs/traits-stubs/trait_types.pyi

```diff
@@ -57,9 +58,9 @@ class Any(_TraitType[_Any, _Any]):
         self,
         default_value: _Any = ...,
         *,
-        factory: _CallableType = ...,
-        args: tuple = ...,
-        kw: dict = ...,
+        factory: _CallableType[..., _Any] = ...,
+        args: _Tuple[_Any, ...] = ...,
+        kw: _DictType[str, _Any] = ...,
```

Comment:
This would seem like a use-case for `ParamSpec` (but that would imply using `typing-extensions` for now).

Reply:
I'm not seeing how to use `ParamSpec` here; any suggestions?

What I'd _like_ to do is something like:
```
P = ParamSpec("P")

class Any(_TraitType[_Any, _Any]):
    def __init__(
        self,
        default_value: _Any = ...,
        *,
        factory: _CallableType[P, _Any] = ...,
        args: P.args = ...,
        kw: P.kwargs = ...,
        **metadata: _Any,
    ) -> None:
        ...
```

But it looks as though `P.args` and `P.kwargs` can't be used this way - they can only be used in the context of `*args` and `**kwargs`.

We already have the dependence on `typing-extensions`, so I'm not adverse to expanding that dependence.